src/librustc/lint/context.rs

```diff
@@ -69,8 +69,9 @@ pub struct LintStore {
 
     /// Map of registered lint groups to what lints they expand to. The first
     /// bool is true if the lint group was added by a plugin. The optional string
-    /// is used to store the new names of deprecated lint group names.
-    lint_groups: FxHashMap<&'static str, (Vec<LintId>, bool, Option<&'static str>)>,
+    /// is used to store the new names of deprecated lint group names and is paired
+    /// with `true` if the deprecation is silent.
+    lint_groups: FxHashMap<&'static str, (Vec<LintId>, bool, Option<(&'static str, bool)>)>,
```

Comment:
I have no stake in this code so feel free to ignore me, but these tuples seem excessive. Wouldn't it be nicer to have a `struct` with named fields?